src/components/Calendar/calendarHeader/single.js

```diff
@@ -0,0 +1,159 @@
+import React, { useRef, useState, useMemo, useEffect, useCallback } from 'react';
+import PropTypes from 'prop-types';
+import { RIGHT_KEY, LEFT_KEY } from '../../../libs/constants';
+import Select from '../../Select';
+import RightIcon from '../icons/rightArrow';
+import LeftIcon from '../icons/leftArrow';
+import StyledControlsContainer from '../styled/controlsContainer';
+import StyledMonthContainer from '../styled/monthContainer';
+import StyledMonth from '../styled/month';
+import StyledArrowButton from '../styled/arrowButton';
+
+export default function SingleCalendarHeader(props) {
+    const {
+        monthLabelId,
+        currentYear,
+        yearsRange,
+        formattedMonth,
+        disablePreviousMonth,
+        disableNextMonth,
+        onPrevMonthClick,
+        onNextMonthClick,
+        onYearChange,
+    } = props;
+
+    const [enableNavKeys, setEnableNavKeys] = useState(false);
+    const refs = [useRef(), useRef(), useRef()];
+    const [focusedItemIndex, setFocusedItemIndex] = useState(-1);
+
+    const handlePrevMonthClick = useCallback(() => {
+        onPrevMonthClick();
+        setFocusedItemIndex(0);
+    }, [onPrevMonthClick]);
+
+    const handleNextMonthClick = useCallback(() => {
+        onNextMonthClick();
+        setFocusedItemIndex(1);
+    }, [onNextMonthClick]);
+
+    const handleYearSelectClick = useCallback(() => {
+        setFocusedItemIndex(2);
+    }, []);
+
+    const keyHandlerMap = useMemo(
+        () => ({
+            [LEFT_KEY]: () => {
+                const delta = disableNextMonth ? 2 : 1;
+                const firstIndx = disablePreviousMonth ? 1 : 0;
+                setFocusedItemIndex(Math.max(focusedItemIndex - delta, firstIndx));
+            },
+            [RIGHT_KEY]: () => {
+                const delta = disableNextMonth ? 2 : 1;
+                setFocusedItemIndex(Math.min(focusedItemIndex + delta, refs.length - 1));
+            },
+        }),
+        [disableNextMonth, disablePreviousMonth, focusedItemIndex, refs.length],
+    );
+
+    const handleKeyDown = useCallback(
+        event => {
+            if (disableNextMonth && disablePreviousMonth) return;
+            if (enableNavKeys) {
+                const { keyCode } = event;
+                if (keyHandlerMap[keyCode]) {
+                    event.preventDefault();
+                    event.stopPropagation();
```

Comment:
Yesterday we discussed this and defined that we have to remove the event.stopPropagation (); because they break the isolate of the components. We have to find another way.